vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -0,0 +1,130 @@
+package ca.on.oicr.gsi.vidarr.server;
+
+import ca.on.oicr.gsi.Pair;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.prometheus.client.Gauge;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+final class PriorityByWorkflow implements ConsumableResource {
+
+  private Integer priority;
+
+  private List acceptedPriorities = Arrays.asList(1, 2, 3, 4);
+
+  private static final class WaitingState {
+
+    private final SortedSet<SimpleEntry<String, Integer>> waiting = new TreeSet<SimpleEntry<String, Integer>>(
+        new PairComparator()){
+
+      @Override
+      public boolean add(SimpleEntry<String, Integer> simpleEntry) {
+        super.remove(this.containsKey(simpleEntry));
+        return super.add(simpleEntry);
+      }
+
+      public SimpleEntry containsKey(SimpleEntry simpleEntry){
+        String entryKey = (String) simpleEntry.getKey();
+        for (SimpleEntry entry : this){
+          if (entry.getKey().equals(entryKey)) {return(entry);}
+        }
+        return null;
+      }
+    };
+  }
+
+  static class PairComparator implements Comparator<SimpleEntry<String, Integer>> {
+
+    @Override
+    public int compare(SimpleEntry<String, Integer> entry1, SimpleEntry<String, Integer> entry2) {
+      return entry1.getValue() - entry2.getValue();
```

Comment:
SimpleEntry already inherits comparingByValue from Map.Entry which returns a Comparator that sorts by natural order: https://docs.oracle.com/javase/8/docs/api/java/util/AbstractMap.SimpleEntry.html